### PR TITLE
Add workspaceEnv to plugin, populate it on sidecar resolving

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Adds an ability to run multiple plugin brokers on plugins provisioning phase. They all be part of a single pod.
Use bare pod instead of deployment for plugin brokering to avoid unwanted restarts of brokers.
Adds Theia plugin broker (which is not merged yet, but should be merged soon).
Allow ChePlugin and plugin brokers (by changing ChePlugins) to provide env vars that are provisioned in all the containers of the workspace. This enables brokers/plugins to share info with each other.

### CODE REVIEW

**Comment by garagatyi** [comment]
ci-build

**Comment by garagatyi** [comment]
ci-build

**Review by skabashnyuk — APPROVED** [review]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java]: Consider moving this to constant

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java]: As for me, brokerConfig variable name would be simpler here

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerStatusListener.java
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerStatusListener.java]: `brokersResult.brokerResult` looks a bit unclear. Please consider renaming to something like `addResult`, `registerResult`

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/BrokersResultTest.java
Comment (sleshchenko) [inline infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/BrokersResultTest.java]: I think it makes sense add `expectedMessagePattern` to clarify that right exception is thrown. WDYT?

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginBase.java
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginBase.java]: I think it would be clearer `variables to set in all machines of the workspace`

**Review by sleshchenko — APPROVED** [review]
LGTM

Please take a look my inline comments and finish tests before merging it =)

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactoryTest.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactoryTest.java]: Was intended not to forget to add unit tests

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/PluginBrokerManagerTest.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/PluginBrokerManagerTest.java]: Was intended not to forget to add unit tests

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties
Comment (garagatyi) [reply assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties]: I think we will rework the approach to something else a bit later. So, the is no sense to improve this approach now.

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java]: ok

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java]: ok

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerStatusListener.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerStatusListener.java]: ok

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/BrokersResultTest.java
Comment (garagatyi) [reply infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/BrokersResultTest.java]: Agree

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginBase.java
Comment (garagatyi) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginBase.java]: I don't want to use the term `machine` here. Looks like we need to get rid of it because it is not clear in containers world. I changed the doc but used containers term instead.

**Comment by garagatyi** [comment]
ci-build

**Comment by garagatyi** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/985/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/985//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11596
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by garagatyi** [comment]
ci-test

**Comment by garagatyi** [comment]
@eclipse/eclipse-che-qa Please, review tests report

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/986/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/986//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11596
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by garagatyi** [comment]
After discussing with @skabashnyuk [here](https://github.com/eclipse/che-plugin-broker/pull/14) we decided not to merge theia broker implementation yet, so I've changed dockerimage in this PR to my personal one where the broker implementation is pushed.

**Comment by garagatyi** [comment]
I analyzed PR CI report and haven't found any signs of my PR spoiling anything. It seems that all the issues are related to known bugs in the project and instability of tests. This is especially interesting since this PR changes components that are used in Workspace.Next flow only, so there is next to zero probability that it interferes with the tests that failed.  
I decide to merge PR because of the reasons mentioned above and approvals from the maintainers.

**Comment by garagatyi** [comment]
I merged with a merge commit instead of the squashing by a mistake. Sorry.

**Pull request merged**
